src/Phinx/Db/Action/Action.php

```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Phinx
+ *
+ * (The MIT license)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated * documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+namespace Phinx\Db\Action;
+
+abstract class Action
```

Comment:
Is there a convention of having an 'Abstract' prefix to abstract classes? Also,by not having a constructor, this is placing the responsibility of populating the $table on the subclass with no type hinting being enforced in this class. Making $table private and having a `protected function setTable(Table $table){$this->table=$table;}` would provide that enforcement of the correct type. Or having a constructor and having various tools (PHPStan for example) to identify when/if the subclass doesn't call the parent class

Reply:
I’ll add a constructor, I don’t want setters in the action classes